### PR TITLE
New atomsWithQuery impl

### PR DESCRIPTION
**Pull request by Aslemammad**

### CODE REVIEW

File: src/atomsWithQuery.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  Variables extends object = OperationVariables,
+  Data = any
+> = QueryOptions<Variables, Data>
+
+type AtomWithQueryAction<
+  Variables extends object = OperationVariables,
+  Data = any
+> = {
+  type: 'refetch'
+  opts?: Partial<QueryArgs<Variables, Data>>
```
Comment (dai-shi) [inline src/atomsWithQuery.ts:21]: This is not used, so it should be removed. (unsupported)

**Review by dai-shi — COMMENTED** [review]
It looks good.

File: src/atomsWithQuery.ts (hunk 1 above)
Comment (Aslemammad) [reply src/atomsWithQuery.ts:21]: Oh, sure! Thanks

File: src/atomsWithQuery.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    getArgs,
+    getClient,
+    (client, args) => {
+      return {
+        subscribe: (observer) => {
+          client
+            .query(args)
+            .then((result) => {
+              observer.next?.(result)
+            })
```
Comment (dai-shi) [inline src/atomsWithQuery.ts:40]: This means we can get the result only once, right?
Maybe it's how apollo works (it's the same before this PR.)

**Review by dai-shi — APPROVED** [review]
The code looks good to me.
Curious if examples/demos work.
Please check failing CI.

File: src/atomsWithQuery.ts (hunk 2 above)
Comment (Aslemammad) [reply src/atomsWithQuery.ts:40]: Tried before, and as far as i remember this is the way it works, Apollo is not as nice as urql in API!

**Review by dai-shi — APPROVED** [review]

**Pull request merged**